### PR TITLE
1942 - adding a new update endpoint

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: web-client/src/presenter/actions/CourtIssuedOrder/isEditingOrderAction.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { state } from 'cerebral';
+
+/**
+ * change path based on if editing an order or creating a new one
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.path the cerebral path used for setting state.submitting
```
Comment (matthopson) [inline web-client/src/presenter/actions/CourtIssuedOrder/isEditingOrderAction.js]: remove `state.submitting`

File: web-client/src/presenter/actions/CourtIssuedOrder/overwriteOrderFileAction.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { state } from 'cerebral';
+
+/**
+ * upload document to s3.
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @returns {object} the next path based on if validation was successful or error
```
Comment (matthopson) [inline web-client/src/presenter/actions/CourtIssuedOrder/overwriteOrderFileAction.js]: is `validation` correct here?

File: web-api/src/cases/updateCourtIssuedOrderToCaseLambda.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    try {
+      const results = await applicationContext
+        .getUseCases()
+        .updateCourtIssuedOrderInteractor({
+          ...JSON.parse(event.body),
+          documentIdToEdit: event.pathParameters.documentId,
+          applicationContext,
+        });
+      applicationContext.logger.info('User', user);
+      applicationContext.logger.info('Results', results);
```
Comment (matthopson) [inline web-api/src/cases/updateCourtIssuedOrderToCaseLambda.js:24]: I have a question about this, but I'll take it to slack.

**Review by matthopson — APPROVED** [review]

File: shared/src/business/useCases/courtIssuedOrder/updateCourtIssuedOrderInteractor.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      ...documentMetadata,
+      relationship: 'primaryDocument',
+      documentId: documentIdToEdit,
+      documentType: documentMetadata.documentType,
+      userId: user.userId,
+      filedBy: user.name,
+    },
+    { applicationContext },
+  );
+  documentEntity.processingStatus = 'complete';
```
Comment (sutt0n) [inline shared/src/business/useCases/courtIssuedOrder/updateCourtIssuedOrderInteractor.js]: small nitpick, but could we put this above in the `Document` constructor?

**Review by sutt0n — APPROVED** [review]

**Comment by matthopson** [comment]
@codyseibert Could we squeeze the swagger docs in here as well?

**Review by matthopson — APPROVED** [review]

**Pull request merged**